### PR TITLE
Fix MSG parse issue with CPTDTXT.MSG

### PR DESCRIPTION
**Pull request by paladine**
MSG parsing code would enter a junk state when encountering a \r\n
immediately after an identifier, but it should allow the transition
to the SPACE state instead. Technically \r\n are space characters
after all.

### CODE REVIEW

**Comment by enusbaum** [comment]
This would undo #481

We might need to add some logic to evaluate if an identifier has been extracted (is the `\r\n` before or after) as the two cases seem distinct.

**Comment by enusbaum** [comment]
Also I'm wondering if we should abstract out the parser to have a similar `Tick()` command (`Step()`?) for parsing, as it would allow us to pass in potentially a string/file and Unit Test state given a specific scenario.

**Comment by enusbaum** [comment]
Re-Wrote the MSG File Loader/Parser. It's much more flexible now and also supports Unit/Integration testing. I tested this PR on `TSGARN`, `WCCMMUD`, and `CPTDAWN` and it loads all the MSG file values properly.

File: MBBSEmu/Module/MsgFile.cs [hunk 1]
```diff
@@ -40,13 +40,37 @@ public MsgFile(IFileUtility fileUtility, string modulePath, string msgName)
...
+        {
+            /// <summary>
+            ///     Data Before Identifier Characters
+            /// </summary>
+            PREKEY,
+
+            /// <summary>
+            ///     Data for Identifier
+            /// </summary>
+            KEY,
```
Comment (paladine) [inline MBBSEmu/Module/MsgFile.cs:53]: comment has identifier, so why not `IDENTIFIER?`

**Pull request merged**